llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -15812,7 +15812,8 @@ SDValue RISCVTargetLowering::PerformDAGCombine(SDNode *N,
           MGN->getMemOperand(), IndexType, MGN->getExtensionType());
 
     if (Index.getOpcode() == ISD::BUILD_VECTOR &&
-        MGN->getExtensionType() == ISD::NON_EXTLOAD) {
+        MGN->getExtensionType() == ISD::NON_EXTLOAD &&
+        isTypeLegal(VT)) {
```

Comment:
I think this is accidentally from https://github.com/llvm/llvm-project/issues/81088#issuecomment-1933416755